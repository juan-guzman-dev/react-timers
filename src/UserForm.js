import React, { useState } from 'react';
import "./UserForm.css";

const UserForm = () => {
    const initialState = {
        username: "",
        email: ""
    }
    const [formData, setFormData] = useState(initialState)
    const [username, setUsername] = useState("")

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        const { username, email } = formData;
        // alert(`Created user, ${username} w/ email ${email}`);
        setFormData(initialState)
        setUsername(username)
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <label htmlFor="username">Username</label>
                <input
                    id="username"
                    type="text"
                    placeholder="username"
                    name="username"
                    value={formData.username}
                    onChange={handleChange}
                />
                <label htmlFor="email">Email</label>
                <input
                    id="email"
                    type="email"
                    placeholder="email"
                    name="email"
                    value={formData.email}
                    onChange={handleChange}
                />
                <button>Create User!</button>
            </form >
            <div>
                {username ? <h4>You are logged in as: {username}</h4> : null}
            </div>
        </>
    )
}

export default UserForm;