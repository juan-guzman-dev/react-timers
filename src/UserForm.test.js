import { render, fireEvent } from '@testing-library/react';
import UserForm from './UserForm';


it("renders without crashing", function () {
    render(<UserForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<UserForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a new user", function () {
    const { getByLabelText, queryByText } = render(<UserForm />);

    // no items yet
    expect(queryByText("hugo2021")).not.toBeInTheDocument();

    const nameInput = getByLabelText("Username");
    const emailInput = getByLabelText("Email");
    const submitBtn = queryByText("Create User!")

    // fill out the form
    fireEvent.change(nameInput, { target: { value: "hugo2012" } });
    fireEvent.change(emailInput, { target: { value: "hugo@email.com" } });
    fireEvent.click(submitBtn);

    // item exists!
    expect(queryByText("You are logged in as: hugo2012")).toBeInTheDocument();
});