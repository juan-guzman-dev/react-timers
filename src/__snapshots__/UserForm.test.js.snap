// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="username"
    >
      Username
    </label>
    <input
      id="username"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      placeholder="email"
      type="email"
      value=""
    />
    <button>
      Create User!
    </button>
  </form>
  <div />
</DocumentFragment>
`;
